### PR TITLE
Fix model parameter updating after fit for compound models

### LINKED ISSUE #1022: [BUG] Specviz: Model fitting no longer updates displayed parameters on fit
@kecnry this may be related to some of your recent work. Previously, fitting a model would update the displayed model parameters in the plugin. Now it appears that the model fits correctly, but the displayed parameter values are not updated. Maybe this got broken in your work rearranging the "fixed" checkboxes somehow?

### PR DESCRIPTION
**Pull request by rosteen**
Fixes #1022 , ended up being very simple. The displayed parameters for models with multiple components now update properly with the fitted values again.

### CODE REVIEW

File: jdaviz/configs/default/plugins/model_fitting/model_fitting.py [hunk 1]
```diff
@@ -159,7 +159,7 @@ def _update_parameters_from_fit(self):
         """Insert the results of the model fit into the component_models"""
         for m in self.component_models:
             name = m["id"]
-            if hasattr(self._fitted_model, name):
+            if hasattr(self._fitted_model, "submodel_names"):
```
Comment (kecnry) [inline jdaviz/configs/default/plugins/model_fitting/model_fitting.py:162]: Looking back at why I made this change for BlackBody, I think you'll need something like the following to handle all cases (current proposed change breaks for adding a BlackBody and Linear component but only fitting the BlackBody component, for example):
```
if name in getattr(self._fitted_model, "submodel_names", []):
```

Comment (rosteen) [reply jdaviz/configs/default/plugins/model_fitting/model_fitting.py:162]: It seems like that case works for me. Are you saying something different than the recording below?

https://user-images.githubusercontent.com/39831871/148075687-042d88e8-abe7-4b50-b54e-33ba7ea29041.mov

Comment (kecnry) [reply jdaviz/configs/default/plugins/model_fitting/model_fitting.py:162]: Ah, it seems that it happens if you have a model with multiple components but also unused components.  See below:

https://user-images.githubusercontent.com/877591/148078170-3eaa3fdb-d075-44f4-acb1-03dc27a96d33.mov

Comment (rosteen) [reply jdaviz/configs/default/plugins/model_fitting/model_fitting.py:162]: Ahh, I see. Thanks for the clarification.

**Comment by rosteen** [comment]
@kecnry I added a check to handle that error.

**Review by kecnry — APPROVED** [review]
Might be worth trying to get coverage for the if-statement in `test_fitting.test_fitting_backend` if its not too difficult, but otherwise looks reasonable to me!

It might also be nice to clear or gray out the values of unused model components, but that's out of scope here.

**Review by pllim — APPROVED** [review]
Parameters update as advertised. Thanks!

**Comment by pllim** [comment]
If you don't want to address Kyle's last comment about testing, feel free to merge.

(Can always chuck moar tests into "technical debts" bin...)

**Comment by rosteen** [comment]
There is a ticket in the UI unit testing epic for adding more unit tests for the model fitting plugin, so I'm going to go ahead and leave that as a separate effort for now and merge this.

**Pull request merged**